File: risk-lib/src/main/java/net/mjduffin/risk/lib/usecase/GameState.java
package net.mjduffin.risk.lib.usecase;

public class GameState {
    private String currentPlayer;
    private String phase;
    public int unitsToPlace;
    public String[] territories;
    public String[] occupyingPlayers;
    public Integer[] units;
    public boolean hasEnded;

    GameState(String currentPlayer, String phase) {
        this.currentPlayer = currentPlayer;
        this.phase = phase;
    }

    public String getCurrentPlayer() {
        return currentPlayer;
    }

    public String getPhase() {
        return phase;
    }
}

File: risk-lib/src/main/java/net/mjduffin/risk/lib/usecase/GameState.kt
package net.mjduffin.risk.lib.usecase

data class GameState(
    val currentPlayer: String,
    val phase: String,
    val unitsToPlace: Int,
    val territories: List<String>,
    var occupyingPlayers: List<String>,
    val units: List<Int>,
    val hasEnded: Boolean = false)